apps/examples/testcase/ta_tc/task_manager/itc/itc_taskmanager_main.c

```diff
@@ -397,54 +545,89 @@ void itc_unicast_handler_2(tm_msg_t *info)
 	}
 }
 
-void itc_unicast_handler(tm_msg_t *info)
+void unicast_handler(tm_msg_t *info)
 {
 	int tm_sample_handle2;
 	int ret;
+	tm_appinfo_t *task_info;
 
 	if (!strncmp((char *)info->msg, TM_START_MSG, strlen(TM_START_MSG))) {
 
 		tm_sample_handle2 = task_manager_register_task("task6", 100, 1024, not_builtin_task, NULL, TM_APP_PERMISSION_ALL, TM_RESPONSE_WAIT_INF);
-		TC_ASSERT_GEQ("task_manager_register_task", tm_sample_handle2, 0);
+		if (tm_sample_handle2 < 0) {
+			printf("ERROR: failed task_manager_register_task in unicast_handler ERR: %d\n", tm_sample_handle2);
+			return;
+		}
 
 		ret = task_manager_start(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
-		TC_ASSERT_EQ_CLEANUP("task_manager_start", ret, OK, unregister_tasks_3(tm_sample_handle2));
+		if (ret != OK) {
+			printf("ERROR: failed task_manager_start in unicast_handler ERR: %d\n", ret);
+			unregister_tasks_3(tm_sample_handle2);
+			return;
+		}
 
 		ret = task_manager_pause(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
-		TC_ASSERT_EQ_CLEANUP("task_manager_pause", ret, OK, unregister_tasks_3(tm_sample_handle2));
+		if (ret != OK) {
+			printf("ERROR: failed task_manager_pause in unicast_handler ERR: %d\n", ret);
+			unregister_tasks_3(tm_sample_handle2);
+			return;
+		}
 
 		ret = task_manager_resume(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
-		TC_ASSERT_EQ_CLEANUP("task_manager_resume", ret, OK, unregister_tasks_3(tm_sample_handle2));
+		if (ret != OK) {
+			printf("ERROR: failed task_manager_resume in unicast_handler ERR: %d\n", ret);
+			unregister_tasks_3(tm_sample_handle2);
+			return;
+		}
 
 		ret = task_manager_restart(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
-		TC_ASSERT_EQ_CLEANUP("task_manager_restart", ret, OK, unregister_tasks_3(tm_sample_handle2));
+		if (ret != OK) {
+			printf("ERROR: failed task_manager_restart in unicast_handler ERR: %d\n", ret);
+			unregister_tasks_3(tm_sample_handle2);
+			return;
+		}
 
 	} else if (!strncmp((char *)info->msg, TM_STOP_MSG, strlen(TM_STOP_MSG))) {
-		tm_appinfo_t *task_info;
 
 		task_info = ((tm_appinfo_t *)task_manager_getinfo_with_pid(pid_task6, TM_RESPONSE_WAIT_INF));
-		TC_ASSERT_NEQ("task_manager_getinfo_with_name", task_info, NULL);
+		if (task_info == NULL) {
+			printf("ERROR: failed task_manager_getinfo_with_pid in unicast_handler\n");
+			return;
+		}
 
 		tm_sample_handle2 = task_info->handle;
 		task_manager_clean_info(&task_info);
-		TC_ASSERT_NEQ("task_manager_clean_infolist", (tm_appinfo_list_t *)&task_info, NULL);
+		if ((tm_appinfo_list_t *)&task_info == NULL) {
+			printf("ERROR: failed task_manager_clean_info in unicast_handler\n");
+			return;
```

Comment:
Should use call "unregister_tasks_3(tm_sample_handle2); " before retrun statement.
"

Reply:
This condition is not required.